src/Persistence/MassTransit.MartenIntegration/Configuration/MartenSagaRepositoryRegistrationExtensions.cs

```diff
@@ -145,12 +155,17 @@ public static void SetMartenSagaRepositoryProvider(this IRegistrationConfigurato
         public static void SetMartenSagaRepositoryProvider(this IRegistrationConfigurator configurator, Func<NpgsqlConnection> connectionFactory,
             Action<StoreOptions> configureOptions)
         {
-            configurator.AddMarten(options =>
-            {
-                options.Connection(connectionFactory);
+            var martenRegistrationPipeline = configurator.AddMarten(
+                options =>
+                {
+                    options.Connection(connectionFactory);
+
+                    configureOptions?.Invoke(options);
+                });
 
-                configureOptions?.Invoke(options);
-            });
+#if NET6_0_OR_GREATER
+            martenRegistrationPipeline.ApplyAllDatabaseChangesOnStartup();
```

Comment:
Obsolete method I know, but without this, database would never be initialised

Reply:
So, I guess from a test/usage perspective, can we assume that developers using Marten 6 are going to understand database creation on their own? Or should we do this to ensure the database is up and configured properly for anyone using the Marten saga repository?

Reply:
In these obsolete methods, I think we need to handle this for them
Anyone using `SetMartenSagaRepositoryProvider()` would have to configure it themselves as they'd be responsible for Marten registration.

I could update the docs to be fully transparent about what happens here, and also to warn about the need to apply this MartenConfigurationExpression on their registration of Marten?

Another option would be to change the method signature on these obsolete methods to take in an optional boolean defaulting to true (so you can opt out), and wrap the call in a check on the boolean.
We may need two booleans tbh - as there are two config expressions that do different things.
This is the code from Marten

```csharp
         /// <summary>
        ///     Adds a hosted service to your .Net application that will attempt to apply any detected database changes before the
        ///     rest of the application starts running
        /// </summary>
        /// <returns></returns>
        public MartenConfigurationExpression ApplyAllDatabaseChangesOnStartup()
        {
            Services.EnsureMartenActivatorIsRegistered();

            Services.ConfigureMarten(opts => opts.ShouldApplyChangesOnStartup = true);

            return this;
        }

        /// <summary>
        ///     Adds a hosted service to your .Net application that will assert that database matches configuration before the
        ///     rest of the application starts running. Prevents the application from starting if database does not match
        ///     configuration.
        /// </summary>
        /// <returns></returns>
        public MartenConfigurationExpression AssertDatabaseMatchesConfigurationOnStartup()
        {
            Services.EnsureMartenActivatorIsRegistered();

            Services.ConfigureMarten(opts => opts.ShouldAssertDatabaseMatchesConfigurationOnStartup = true);

            return this;
        }
```
Both hosted services serve different purposes.